### ISSUE TITLE
Error when using CTE and self join

### ISSUE THREAD

**Issue by matbur** [sqlc-dev/sqlc#2132]
### Version

1.17.2

### What happened?

When using CTE (WITH clause) in query and self join, sqlc doesn't see any columns. When using join from another table/view it works as expected.

### Relevant log output

```shell
sqlc generate failed.

# package db
query.sql:6:5: column "id" does not exist
```

### SQL queries

```sql
-- name: Q :many
WITH
    items1 AS (SELECT 'id'::TEXT AS id, 'name'::TEXT AS name),
    items2 AS (SELECT 'id'::TEXT AS id, 'name'::TEXT AS name)
SELECT
    i1.id AS id1,
    i2.id AS id2
FROM
    items1 i1
        JOIN items1 i2 ON 1 = 1;
        -- change ^ above items1 to items2 to fix
```

### Configuration

```yaml
{
  "version": "1",
  "packages": [
    {
      "path": "db",
      "engine": "postgresql",
      "schema": "query.sql",
      "queries": "query.sql"
    }
  ]
}
```

### Playground URL

https://play.sqlc.dev/p/af3970dbc2ab065393d42b65a2bedf37b0d73734214ca9a42579829dfb8e6de7

### What database engines are you using?

PostgreSQL

### What type of code are you generating?

Go

**Comment by kyleconroy** [comment]
This is fixed in [v1.23.0](https://docs.sqlc.dev/en/latest/reference/changelog.html#v1-23-0) by enabling the [database-backed query analyzer](https://docs.sqlc.dev/en/latest/howto/generate.html#enhanced-analysis-with-managed-databases). We added a [test case](https://github.com/sqlc-dev/sqlc/tree/main/internal/endtoend/testdata/cte_join_self/postgresql/pgx) for this issue so it won’t break in the future.

You can play around with the [working example on the playground](https://play.sqlc.dev/p/d846ab72149e9a7eec0fb60809f379e9d79e3d80aacef1962f29c75167cd4c08)

**Comment by matbur** [comment]
Thank you for fixing this! Now it works perfectly.
